NuRadioMC/simulation/simulation.py

```diff
@@ -401,752 +1339,325 @@ def __init__(self, inputfilename,
+                                                min_efield_amplitude=float(self._config['speedup']['min_efield_amplitude']) * self._Vrms_efield_per_channel[sid][channel_id],
+                                                time_logger=self.__time_logger)
+                        # skip to next channel if the efield is below the speed cut
+                        if len(sim_station.get_electric_fields()) == 0:
+                            logger.info(f"Eventgroup {event_group.get_run_number()} Station {sid} channel {channel_id:02d} has {len(sim_station.get_electric_fields())} efields, skipping to next channel")
                             continue
 
-                        n = self._raytracer.get_number_of_solutions()
-                        for iS in range(n):  # loop through all ray tracing solution
-                            # skip individual channels where the viewing angle difference is too large
-                            # discard event if delta_C (angle off cherenkov cone) is too large
-                            if np.abs(delta_Cs[iS]) > self._cfg['speedup']['delta_C_cut']:
-                                logger.debug('delta_C too large, ray tracing solution unlikely to be observed, skipping event')
+                        # applies the detector response to the electric fields (the antennas are defined
+                        # in the json detector description file)
+                        apply_det_response_sim(sim_station, self._det, self._config, self.detector_simulation_filter_amp,
+                                            event_time=self._evt_time, time_logger=self.__time_logger,
+                                            detector_simulation_part1=self.detector_simulation_part1)
+                        logger.debug(f"adding sim_station to station {sid} for event group {event_group.get_run_number()}, channel {channel_id}")
+                        station.add_sim_station(sim_station)  # this will add the channels and efields to the existing sim_station object
+                        for evt in output_buffer[sid].values():
+                            trace_start_time = evt.get_station().get_channel(channel_ids[0]).get_trace_start_time()
+                            # determine the trigger that was used to determine the readout window
+                            trigger_name = None
+                            trigger_time = None
+                            pre_trigger_time = None
+                            for trigger in evt.get_station().get_triggers().values():
+                                if trigger.get_pre_trigger_times() is not None:
+                                    pre_trigger_time = trigger.get_pre_trigger_times()
+                                    trigger_time = trigger.get_trigger_time()
+                                    trigger_name = trigger.get_name()
+                                    break
```

Comment:
don't we have to select the min trigger time and not the trigger time of the first trigger with a not None pre trigger time?

Reply:
it is doing that. 
# we need to select the trigger that was used to determine the readout window, which is the one that has the
                                # `pre_trigger_times` set. All other triggers will return None. We can double check that by testing that the
                                #  trigger_time is the same as start time as the trace
I added a comment to clarify that. This is maybe not ideal but it is that way how we do things at the moment. If we want to change that, we also need to change the way the triggerTimeAdjuster works.

Reply:
> we also need to change the way the triggerTimeAdjuster works

I think we should do that :)